From f41999329064d4f46bd8d144cda458da787a9c45 Mon Sep 17 00:00:00 2001
From: ZoliN <zltnngy@gmail.com>
Date: Sun, 1 Jun 2014 22:16:02 +0200
Subject: [PATCH] libstagefright: Fix video encoder input buffer

Ported from cm10, needed to avoid the encoder eating up all input buffers,
which leaves no buffer left for the camera sometimes

Change-Id: I0432cd87a9defd08725ac8831d7376edb98d7e18
---
 media/libstagefright/OMXCodec.cpp |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index 7073251..9e7b092 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -3645,11 +3645,12 @@ void OMXCodec::drainInputBuffers() {
                 continue;
             }
 
-#if defined OMAP_ENHANCEMENT && defined TARGET_OMAP3
+#if defined(OMAP_ENHANCEMENT) && defined(TARGET_OMAP3) || defined(SEMC_ICS_CAMERA_BLOB)
             if (mIsEncoder && mIsVideo && (i == 4)) {
                 break;
             }
 #endif
+
             if (!drainInputBuffer(info)) {
                 break;
             }
-- 
1.7.9.5

